.github/workflows/deploy.yml

```diff
@@ -0,0 +1,90 @@
+name: Deploy to Demo and Test
+on:
+  push:
+    branches:
+      - master
+      - develop
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v1
+      - name: Setup node
+        uses: actions/setup-node@v1
+        with:
+          node-version: "12.x"
+      
+      # Build for demo
+      - name: Build and publish demo docker image 
+        if: github.ref == 'refs/heads/master'
+        uses: elgohr/Publish-Docker-Github-Action@master
+        with:
+          name: demo-storefrontcloud-io/vue-storefront-api:${{ github.sha }}
+          registry: registry.storefrontcloud.io
+          username: ${{ secrets.CLOUD_USERNAME }}
+          password: ${{ secrets.CLOUD_PASSWORD }}
+          dockerfile: dev/docker/Dockerfile
+          buildoptions: "--compress"
+      
+      # Build for test
+      - name: Build and publish test docker image
+        if: github.ref == 'refs/heads/develop'
+        uses: elgohr/Publish-Docker-Github-Action@master
+        with:
+          name: test-storefrontcloud-io/vue-storefront-api:${{ github.sha }}
+          registry: registry.storefrontcloud.io
+          username: ${{ secrets.CLOUD_USERNAME }}
+          password: ${{ secrets.CLOUD_PASSWORD }}
+          dockerfile: dev/docker/Dockerfile
+          buildoptions: "--compress"
+
+  deploy:
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+      - uses: chrnorm/deployment-action@releases/v1
+        name: Create GitHub deployment for ${GITHUB_REF##*/}
+        if: github.ref == 'refs/heads/master' || github.ref == 'refs/heads/develop'
+        id: deployment
+        with:
+          token: "${{ github.token }}"
+          target_url: https://${GITHUB_REF##*/}.europe-west1.gcp.storefrontcloud.io
+          environment: production
+          initial_status: in_progress
+
+      # Deploy on test/demo depending on the branch that triggered the workflow
+      - name: Deploy on ${GITHUB_REF##*/}.europe-west1.gcp.storefrontcloud.io
+        if: github.ref == 'refs/heads/master' || github.ref == 'refs/heads/develop'
```

Comment:
This if unnecessary, this job is always running for one of those branches because you have
```
on:
  push:
    branches:
      - master
      - develop
```